### PR TITLE
Stop reconciliation and clustering

### PR DESCRIPTION
**Pull request by d-kuro**
refs: 
https://github.com/cybozu-go/moco/issues/572
https://github.com/cybozu-go/moco/issues/564

Add the enhancement to stop clustering and reconciling MySQLCluster based on the following design document.
https://github.com/cybozu-go/moco/blob/main/docs/designdoc/clustering_stop.md

### CODE REVIEW

File: e2e/stop_test.go [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+		Expect(reconcileMf).NotTo(BeNil())
+		reconcileMetric := findMetric(reconcileMf, map[string]string{"namespace": "stop", "name": "test"})
+		Expect(reconcileMetric).NotTo(BeNil())
+		Expect(reconcileMetric.GetGauge().GetValue()).To(BeNumerically("==", 1))
+
+		healthyMf := mfs["moco_cluster_healthy"]
+		Expect(healthyMf).NotTo(BeNil())
+		healthyMetric := findMetric(healthyMf, map[string]string{"namespace": "stop", "name": "test"})
+		Expect(healthyMetric).NotTo(BeNil())
+		Expect(math.IsNaN(healthyMetric.GetGauge().GetValue())).To(BeTrue())
```
Comment (kmdkuk) [inline e2e/stop_test.go:323]: Is it too redundant to test that `moco_cluster_available`, `moco_cluster_ready_replicas`, `moco_cluster_errant_replicas` are also `NaN`?

File: e2e/stop_test.go [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+			if currentPrimaryIndex == cluster.Status.CurrentPrimaryIndex {
+				return errors.New("failover not yet executed")
+			}
+
+			for _, cond := range cluster.Status.Conditions {
+				if cond.Type != mocov1beta2.ConditionHealthy {
+					continue
+				}
+				if cond.Status == metav1.ConditionTrue {
+					currentPrimaryIndex = cluster.Status.CurrentPrimaryIndex
```
[the remark this answers is not in the document]
Comment (d-kuro) [reply e2e/stop_test.go]: https://github.com/cybozu-go/moco/pull/578/commits/6ee01df82339a4f4eef1cc61797e92c0b2f924b9

You're right, it was unnecessary, so I deleted it.

File: cmd/kubectl-moco/cmd/stop.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+
+	mocov1beta2 "github.com/cybozu-go/moco/api/v1beta2"
+	"github.com/cybozu-go/moco/pkg/constants"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/types"
```
[the remark this answers is not in the document]
Comment (d-kuro) [reply cmd/kubectl-moco/cmd/stop.go]: Fixed: https://github.com/cybozu-go/moco/pull/578/commits/34ba8ed2233743ee45886e9284c6bcf642dcabd3

File: e2e/stop_test.go (hunk 1 above)
Comment (d-kuro) [reply e2e/stop_test.go:322]: https://github.com/cybozu-go/moco/pull/578/commits/fb35e368e51a2ad9c8b4a48ab56747f2fa048777

I have added additional tests regarding metrics.

**Comment by d-kuro** [comment]
I have added functionality to display messages when stopping or resuming clustering and reconciliation using the `kubectl moco` command.
https://github.com/cybozu-go/moco/pull/578/commits/41076822eb1b6be4a3c8e4022ae29c81ca849656

**Comment by kmdkuk** [comment]
LGTM
However, I would like to wait for a review of zoetrope.

**Comment by d-kuro** [comment]
@zoetrope 
Thanks for the comment. I have fixed it with the following commit.
https://github.com/cybozu-go/moco/pull/578/commits/75572b773eea65d41c2bc6554013cdeef5a82ac4

**Review by zoetrope — APPROVED** [review]
LGTM

**Pull request merged**